source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidRealTimeMultiplayerClient.cs

```diff
@@ -34,111 +56,689 @@ public AndroidRealTimeMultiplayerClient(AndroidJavaObject account)
+            var roomStatus = GetRoomStatus();
+            if (roomStatus != RoomStatus.Active && roomStatus != RoomStatus.Connecting)
+            {
+                OurUtils.Logger.d("Sending message is not allowed in this state.");
+                return;
+            }
+
+            if (GetParticipant(participantId) == null)
+            {
+                OurUtils.Logger.e("Attempted to send message to unknown participant " + participantId);
+                return;
+            }
+
+            string roomId = mRoom.Call<string>("getRoomId");
+            if (reliable)
+            {
+                using (mRtmpClient.Call<AndroidJavaObject>("sendReliableMessage", data, roomId, participantId,
+                    /* callback= */ null))
+                    ;
+                return;
+            }
+
+            using (mRtmpClient.Call<AndroidJavaObject>("sendUnreliableMessage", data, roomId, participantId)) ;
         }
 
         public void SendMessage(bool reliable, string participantId, byte[] data, int offset, int length)
         {
-            // Task<Integer> sendReliableMessage(@NonNull byte[] messageData, @NonNull String roomId, @NonNull String recipientParticipantId, @Nullable ReliableMessageSentCallback callback)
-            // Task<Void> sendUnreliableMessage(@NonNull byte[] messageData, @NonNull String roomId, @NonNull String recipientParticipantId)
+            SendMessage(reliable, participantId, Misc.GetSubsetBytes(data, offset, length));
         }
 
         public List<Participant> GetConnectedParticipants()
         {
-            return null;
+            List<Participant> result = new List<Participant>();
+            foreach (Participant participant in GetParticipantList())
+            {
+                if (participant.IsConnectedToRoom)
+                {
+                    result.Add(participant);
+                }
+            }
+
+            return result;
+        }
+
+        private List<Participant> GetParticipantList()
+        {
+            if (mRoom == null)
+            {
+                return new List<Participant>();
+            }
+
+            List<Participant> participants = new List<Participant>();
+            using (var participantsObject = mRoom.Call<AndroidJavaObject>("getParticipants"))
+            {
+                int size = participantsObject.Call<int>("size");
+                for (int i = 0; i < size; i++)
+                {
+                    using (var participant = participantsObject.Call<AndroidJavaObject>("get", i))
+                    {
+                        participants.Add(AndroidJavaConverter.ToParticipant(participant));
+                    }
+                }
+            }
+
+            return participants;
         }
 
         public Participant GetSelf()
         {
+            if (GetRoomStatus() != RoomStatus.Active)
```

Comment:
No need to check it as it will be the same with empty result returned by GetParticipantList